Algorithm.Framework/Portfolio/ReturnsSymbolData.cs

```diff
@@ -38,26 +44,54 @@ public ReturnsSymbolData(Symbol symbol, int lookback, int period)
             _roc.Updated += OnRateOfChangeUpdated;
         }
 
-        public Dictionary<DateTime, double> Returns => _window.Select(x => new { Date = x.EndTime, Return = (double)x.Value }).ToDictionary(r => r.Date, r => r.Return);
+        /// <summary>
+        /// Historical returns
+        /// </summary>
+        public Dictionary<DateTime, double> Returns => _window.ToDictionary(x => x.EndTime, x => (double) x.Value);
 
+        /// <summary>
+        /// Adds an item to this window and shifts all other elements
+        /// </summary>
+        /// <param name="time">The time associated with the value</param>
+        /// <param name="value">The value to use to update this window</param>
         public void Add(DateTime time, decimal value)
         {
+            if (_window.Samples > 0 && _window[0].EndTime == time)
+            {
+                return;
+            }
+
             var item = new IndicatorDataPoint(_symbol, time, value);
             _window.Add(item);
         }
 
+        /// <summary>
+        /// Updates the state of the RateOfChange with the given value and returns true
+        /// if this indicator is ready, false otherwise
+        /// </summary>
+        /// <param name="time">The time associated with the value</param>
+        /// <param name="value">The value to use to update this indicator</param>
+        /// <returns>True if this indicator is ready, false otherwise</returns>
```

Comment:
Does this require an equivalent timestamp check as above?

Reply:
No. Since `_roc` is an indicator, it already has a timestamp checked in [IndicatorBase.cs#L78](https://github.com/QuantConnect/Lean/blob/master/Indicators/IndicatorBase.cs#L78).